wpimath/src/main/java/edu/wpi/first/math/MathUtil.java

```diff
@@ -97,4 +97,15 @@ public static double angleModulus(double angleRadians) {
   public static double interpolate(double startValue, double endValue, double t) {
     return startValue + (endValue - startValue) * MathUtil.clamp(t, 0, 1);
   }
+
+  /**
+   * Adds a scalar magnitude to a value.
```

Comment:
Could you elaborate on the semantics here? Is it saying that if the value is negative, it becomes more negative by the scalar magnitude, and if it's positive, it becomes more positive? The docs here could also be interpreted as `addScalar(-4, 1)` giving `-3`.

What happens if the user enters a value of zero, or negative zero?

Reply:
Your initial interpretation is correct.  What is a concise way to say that or was the above what you would suggest?  

The test in Java for -0 failed and I tested it and it didn't flip the sign.  I don't understand the internals so I just decided to remove that test.

Reply:
Yea, pretty much that. We should give the function defined behavior for value = 0 though.

Reply:
You mean specifically call it out?  Is that an expected use case?  I guess I don't know what it should or what we want it to do and how that math is handled at a low level.

Reply:
I'm not sure what the intended use case is for this function in general, actually. In what specific situations would it be used?